src/backend/joanie/core/models/courses.py

```diff
@@ -233,27 +233,34 @@ def signature_backend_references_to_sign(
                 submitted_for_signature_on__isnull=False,
                 student_signed_on__isnull=False,
                 order__organization=self,
-            ).values_list("signature_backend_reference", flat=True)
+            ).values_list("id", "signature_backend_reference")
         )
         if contracts_ids and len(contracts_to_sign) != len(contracts_ids):
             raise exceptions.NoContractToSignError(
                 "Some contracts are not available for this organization."
             )
-        return contracts_to_sign
+
+        contract_ids = [contract[0] for contract in contracts_to_sign]
+        references = [contract[1] for contract in contracts_to_sign]
+
+        return (references, contract_ids)
 
     def contracts_signature_link(
         self, user: User, contracts_ids: list[str] | None = None
     ):
         """
         Retrieve a signature invitation link for all available contracts.
         """
-        references = self.signature_backend_references_to_sign(contracts_ids)
+        (references, ids) = self.signature_backend_references_to_sign(contracts_ids)
```

Comment:
If the last suggestion is approved:
```suggestion
        ids, references = self.signature_backend_references_to_sign(contracts_ids)
```